testing/integration/upgrade_standalone_fips_test.go

```diff
@@ -0,0 +1,171 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+//go:build integration
+
+package integration
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/upgrade"
+	atesting "github.com/elastic/elastic-agent/pkg/testing"
+	"github.com/elastic/elastic-agent/pkg/testing/define"
+	"github.com/elastic/elastic-agent/pkg/version"
+	"github.com/elastic/elastic-agent/testing/upgradetest"
+)
+
+// This file contains upgrade tests that pertain to FIPS-capable Agent artifacts.
+
+// TestStandaloneUpgradeFIPStoFIPS ensures that upgrading a FIPS-capable Agent
+// results in the new (post-upgrade) Agent to also be FIPS-capable.
+func TestStandaloneUpgradeFIPStoFIPS(t *testing.T) {
+	define.Require(t, define.Requirements{
```

Comment:
How is the runtime dependency on OpenSSL in the FIPS agent being satisfied here? Since there is no FIPS specific requirement here. Do we just always run in VMs with the FIPS OpenSSL available?

Also, I would have expected this test to be limited only to Linux since we don't support FIPS on the other platforms.

Reply:
Yeah, good catch.  There is nothing in this (or the below) test that would ensure that they exercise the FIPS Agent in an environment that has OpenSSL with the FIPS provider.  Perhaps we need a new field in `define.Requirements` so tests can identify themselves as needing to be run only in environments that have OpenSSL with the FIPS provider.  I will add something along those lines in this PR.

And yes, the OS in `define.Requirements` should also be restricted to just Linux.  I will make that change as well.  Thank you!

Reply:
> Yeah, good catch. There is nothing in this (or the below) test that would ensure that they exercise the FIPS Agent in an environment that has OpenSSL with the FIPS provider. Perhaps we need a new field in define.Requirements so tests can identify themselves as needing to be run only in environments that have OpenSSL with the FIPS provider. I will add something along those lines in this PR.

Actually, I think this will be a substantial enough change to the integration testing framework and CI scripts that I created a separate issue to track it: https://github.com/elastic/ingest-dev/issues/5491.

Reply:
Thanks I would expect the tests to fail until that is resolved, just sanity checking and it seems to be the case as the expected error is in the FIPS upgrade tests:

```
could not unmarshal agent version output: error: exit status 2, output: panic: opensslcrypto: FIPS mode requested (requirefips tag set) but not available in OpenSSL 3.0.13 30 Jan 2024
```

If this weren't happening something more fundamental would be wrong (FIPS tests don't actually run FIPS agents).

Reply:
PR for allowing integration tests to declare themselves as needing a FIPS-configured environment is up: https://github.com/elastic/elastic-agent/pull/8083